### ISSUE TITLE
Problem with naming of binutils tools

### ISSUE THREAD

**Issue by pablochacin** [thepowersgang/rust-barebones-kernel#9]
Hi

When running make I get the following error

```
x86_64-none-elf-as  -o .obj/amd64/start.o arch/amd64/start.S
make: x86_64-none-elf-as: Command not found
Makefile:75: recipe for target '.obj/amd64/start.o' failed
make: *** [.obj/amd64/start.o] Error 127
```

I realized this is the name given in the Makefile to as:

```
ifeq ($(ARCH),amd64)
    LD := x86_64-none-elf-ld
    AS := x86_64-none-elf-as
    OBJDUMP := x86_64-none-elf-objdump
    OBJCOPY := x86_64-none-elf-objcopy
else ifeq ($(ARCH),x86)
    LD := i686-elf-ld
    AS := i686-elf-as
    OBJDUMP := i686-elf-objdump
else
     $(error Unknown architecture $(ARCH))
endif
```

If I change the Makefile ot set the name of the tools, the make process continues normally 

```
LD := ld
AS := as
OBJDUMP := objdump
OBJCOPY := objcopy
```

**Comment by thepowersgang** [comment]
yes, as written it requires a bare cross compiler (As is standard practice for OSDev)

**Comment by aristarh2704** [comment]
It's very easy to define a host triple using config.guess

**Comment by thepowersgang** [comment]
Hopefully fixed in df056760c3bc5948f513098315fb9239cf2c35e3 - The target triple can now be overridden with an environment variable (to use the system linker/as)
